### DISCUSSION TITLE
Navigating from a click handler

### DISCUSSION THREAD

**Discussion by alexandremjacques** [unpoly/unpoly#389] [Q&A] [answered]
Hi,

What should be the recommended approach to navigate programmatically? E.g. I have a calendar (built with a regular `<table>`) that should navigate to a day view by clicking on its `<td>` elements.

Since I don't have an `<a>` tag inside the `<td>`s, I'm relying on `up.on('click', '.calendar-day', ...)` and I cannot (or at least I can't see how) to use `up.follow`. Is there a way no navigate direcly to an URL?

TIA

**Comment by adam12** [comment]
I'm not sure I follow. Does `up.follow` not navigate for you?

I'd probably figure out a way to have a regular `<a>` link available inside that table, if not to just support fallback if JS isn't available. You can either define it as a block element that uses the entire table or use `[up-expand]`. If this isn't helpful, perhaps you could make a Glitch using our [starter repo](https://glitch.com/edit/#!/empty-unpoly-project) to demonstrate?

**Reply by alexandremjacques** [reply]
Hi, thanks for taking the time:

https://glitch.com/edit/#!/chrome-playful-fang

Clicking works fine and I can retrieve the data needed. Navigating to the next page is my issue.

**Comment by adam12** [comment, marked as answer]
See my remix. https://glitch.com/edit/#!/shiny-chemical-mozzarella?path=scripts.js%3A4%3A74

I'd still try to handle this with a `<a>` elements for the graceful fallback, IMHO, but hopefully this gets you unblocked.

**Reply by alexandremjacques** [reply]
If the design fits, I'll take the `<a>` element path.

Thanks, it surely helps.
